### PR TITLE
[Tweaks] Firebird buffs

### PR DESCRIPTION
**Pull request by DeityLink**
- When the Firebird drives into an airlock, it'll go through without having to wait for the airlock to fully open (the airlock will still open, and the firebird won't go through at all if the driver doesn't have access)
- Vehicles can now push un-anchored structures (closets, crates, etc)
- Firebird can drive into tables/racks and break them.
- You DON'T want to drive into a welding fuel tank.

### CODE REVIEW

File: code/game/objects/structures/vehicles/wizmobile.dm [hunk 1]
```diff
@@ -69,6 +69,36 @@
...
+	if(istype(obstacle, /obj/structure/table))
+		var/obj/structure/table/T = obstacle
+		T.destroy()
+
+	if(istype(obstacle, /obj/structure/rack))
+		new /obj/item/weapon/rack_parts(obstacle.loc)
+		qdel(obstacle)
+
+	if(istype(obstacle, /obj/structure/reagent_dispensers/fueltank))//MWAHAHAHA
+		obstacle.ex_act(1)
```
Comment (SkowronX) [inline code/game/objects/structures/vehicles/wizmobile.dm]: woah hamburgers

Comment (clusterfack) [inline code/game/objects/structures/vehicles/wizmobile.dm]: You realize Deity it is of critical importance you put this kind of shit into your opening message right

Comment (dylanstrategie) [inline code/game/objects/structures/vehicles/wizmobile.dm]: I came back just to say that

Don't be me :^)

**Comment by IratePirate** [comment]
I don't think it should smash windows. There is nothing fun about a Wizard driving around and venting the entire station within 5 minutes. Plus you could replace the Firebird space fire effect with something custom made perhaps, it'd look much better.

**Comment by NigglyWiggly** [comment]
This probably isn't the place to suggest it, but couldn't we just make the firebird as powerful as it used to be, but a lot more expensive, and make a magic boom cost the same as the firebird currently does and have it act the same as the nerfed one?

Also if nothing else please make it so you can go through the rock walls of the asteroid with the firebird, I want the ones you can find while mining to be not useless again

**Comment by clusterfack** [comment]
Firebird phasing was just abused to hide within walls and shoot people niggly.

File: code/game/objects/structures/vehicles/wizmobile.dm (hunk 1 above)
Comment (ComicIronic) [inline code/game/objects/structures/vehicles/wizmobile.dm]: Not to mention PUTTING IT IN THE CHANGELOG

File: code/game/objects/structures/vehicles/wizmobile.dm [hunk 2]
```diff
@@ -69,6 +69,36 @@
...
+		G.density = 0
+		getFromPool(/obj/item/stack/rods, get_turf(G.loc))
+	*/
+
+	if(istype(obstacle, /obj/structure/table))
+		var/obj/structure/table/T = obstacle
+		T.destroy()
+
+	if(istype(obstacle, /obj/structure/rack))
+		new /obj/item/weapon/rack_parts(obstacle.loc)
```
Comment (ComicIronic) [inline code/game/objects/structures/vehicles/wizmobile.dm]: You can do the same as you did for tables for racks.
They also have a destroy() proc that generates their parts.

**Comment by ririchiyo** [comment]
Wouldn't it just be better to use a firebird_act() proc in this case?

**Comment by DeityLink** [comment]
@IratePirate It doesn't smash windows. Everyone agreed it was a bad idea, so I commented it out.

@NigglyWiggly >as powerful as it used to be

No Niggly, no, there was nothing fun about that wallhack.

**Comment by FudgePucker** [comment]
Oh man being a depressurizing wizard will be fun.
Take gem hardsuit, firebird, fireball, magic missle/faggot instagib spell and remove all the air from the station.

**Comment by DeityLink** [comment]
@FudgePucker 
>depressurizing wizard

what?

**Comment by FudgePucker** [comment]
Ride around busting open windows to space, fireballing open holes into space, destroy the air tank.
GG everyone not in a hard suit.

**Comment by DeityLink** [comment]
@FudgePucker 

>Ride around fireballing open holes into space

Can't they already do that? In case you missed many of my statements in both the thread and these comments I've removed the ability for the firebird to break windows/grilles.

**Comment by FudgePucker** [comment]
O-oh...
My bad.

**Comment by DeityLink** [comment]
It's been over a month, can I get a verdict before some conflict arise? Reminder, this does NOT allow the firebird to break windows.

**Comment by DeityLink** [comment]
bump

**Comment by d3athrow** [comment]
put the collision shit into bump proc for the offending structures.

**Pull request merged**